cmake/HPX_AddModule.cmake

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) 2019 Auriane Reverdell
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+function(add_hpx_module name)
+  # Retrieve arguments
+  set(options DEPRECATION_WARNINGS)
+  set(one_value_args COMPATIBILITY_HEADERS)
+  set(multi_value_args SOURCES HEADERS COMPAT_HEADERS DEPENDENCIES CMAKE_SUBDIRS)
+  cmake_parse_arguments(${name} "${options}" "${one_value_args}" "${multi_value_args}" ${ARGN})
+
+  # Compatibility header off is not specified
+  if (NOT ${name}_COMPATIBILITY_HEADERS)
+      set(${name}_COMPATIBILITY_HEADERS OFF)
+  endif()
+
+  # Main directories of the module
+  set(SOURCE_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/src")
+  hpx_debug("Add module ${name}: SOURCE_ROOT: ${SOURCE_ROOT}")
+  set(HEADER_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/include")
+  hpx_debug("Add module ${name}: HEADER_ROOT: ${HEADER_ROOT}")
+  set(COMPAT_HEADER_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/include_compatibility")
+  hpx_debug("Add module ${name}: COMPAT_HEADER_ROOT: ${COMPAT_HEADER_ROOT}")
+
+  string(TOUPPER ${name} name_upper)
+
+  # HPX options
+
+  hpx_option(HPX_${name_upper}_WITH_TESTS
+    BOOL
+    "Build HPX ${name} module tests. (default: ON)"
+    ON ADVANCED
```

Comment:
Shouldn't this use `${HPX_WITH_TESTS}` for the default value (see for instance: https://github.com/STEllAR-GROUP/hpx/blob/master/libs/config/CMakeLists.txt#L21)?